community/kernel/src/main/java/org/neo4j/kernel/api/cursor/PropertyItem.java

```diff
@@ -43,26 +41,56 @@
     boolean booleanValue();
 
     /**
+     * @return the long value of the current property
+     * @throws IllegalStateException if current property is not a long number
+     */
+    byte byteValue();
+
```

Comment:
This violates the type system we are moving to, with long, double being the only ways to see numbers. Please remove again.